runlog.c

```diff
@@ -312,6 +312,7 @@ run_add_record(
         int            eoln_type,
         int            variant,
         int            is_hidden,
+        int            upsolving,
```

Comment:
Maybe change runlog_state_t to serve_state_t and just make state = cs->runlog_state ?